backend/models/sequence.js

```diff
@@ -144,12 +264,43 @@ Sequence.getList = async (account, model, branch, revision, cleanResponse = fals
 };
 
 Sequence.updateSequence = async (account, model, sequenceId, data) => {
-	if (!data || !data.name || !utils.isString(data.name) || data.name === ""  || data.name.length >= 30) {
+	const toSet = {};
+
+	if (!data) {
 		throw responseCodes.INVALID_ARGUMENTS;
 	}
 
+	if (data.name) {
+		if (!utils.isString(data.name) || data.name === "" || data.name.length >= 30) {
+			throw responseCodes.INVALID_ARGUMENTS;
+		}
+
+		toSet.name = data.name;
+	}
+
+	if (data.revId) {
+		const history = await History.getHistory(account, model, undefined, data.revId);
+
+		if (!history) {
+			throw responseCodes.INVALID_TAG_NAME;
+		}
+
+		toSet.rev_id = history._id;
+	}
+
+	if (data.frames) {
+		const customSequence = await db.findOne(account, sequenceCol(model),
+			{_id: utils.stringToUUID(sequenceId), customSequence: true});
+
+		if (!customSequence) {
+			throw responseCodes.SEQUENCE_READ_ONLY;
+		}
+
+		toSet.frames = await handleFrames(account, model, sequenceId, data.frames);
+	}
+
 	const { result } = await db.update(account, sequenceCol(model),
-		{_id: utils.stringToUUID(sequenceId)}, {$set: {name: data.name}});
+		{_id: utils.stringToUUID(sequenceId)}, {$set: toSet});
```

Comment:
we should check if sequence exist on the top, otherwise (esp with the frames stuff that would also potentially write things into the database), we could end up with a lot of zombie data.